hivemq-edge/src/frontend/src/extensions/datahub/components/nodes/OperationNode.tsx

```diff
@@ -1,34 +1,44 @@
-import { FC } from 'react'
+import { FC, useMemo } from 'react'
 import { useTranslation } from 'react-i18next'
-import { Handle, NodeProps, Position } from 'reactflow'
+import { NodeProps, Position } from 'reactflow'
 import { HStack, Text, VStack } from '@chakra-ui/react'
 
-import { DataHubNodeType, OperationData } from '../../types.ts'
-import { styleSourceHandle } from '../../utils/node.utils.ts'
+import { DataHubNodeType, FunctionDefinition, OperationData } from '../../types.ts'
 import NodeIcon from '../helpers/NodeIcon.tsx'
 import { NodeWrapper } from './NodeWrapper.tsx'
+import { CustomHandle } from '@/extensions/datahub/components/nodes/CustomHandle.tsx'
 
 export const OperationNode: FC<NodeProps<OperationData>> = (props) => {
   const { t } = useTranslation('datahub')
   const { data, id, type } = props
   const { action } = data
 
+  const model = useMemo(() => {
+    if (!action) return undefined
+
+    return typeof action === 'string'
+      ? ({
+          functionId: action,
+        } as FunctionDefinition)
+      : action
+  }, [action])
+
   return (
     <>
       <NodeWrapper route={`node/${DataHubNodeType.OPERATION}/${id}`} {...props}>
         <HStack>
           <NodeIcon type={DataHubNodeType.OPERATION} />
-          <Text> {t('workspace.nodes.type', { context: type })}</Text>
+          <Text data-testid={'node-title'}> {t('workspace.nodes.type', { context: type })}</Text>
           <VStack>
-            <Text>{action?.functionId || '< none >'}</Text>
+            <Text data-testid={'node-model'}>{model?.functionId || t('error.noSet.select')}</Text>
```

Comment:
suggestion: sorry for annoying you with this one :D

```suggestion
            <Text data-testid="node-model">{model?.functionId || t('error.noSet.select')}</Text>
```